test/controllers/webhooks/repo_updates_controller_test.rb

```diff
@@ -0,0 +1,49 @@
+require_relative './base_test_case'
+
+class Webhooks::RepoUpdatesControllerTest < Webhooks::BaseTestCase
+  test "create should return 500 when signature is invalid" do
+    payload = {
+      ref: 'refs/heads/master',
+      repo: { name: 'csharp' }
+    }
+
+    invalid_headers = headers(payload)
+    invalid_headers['HTTP_X_HUB_SIGNATURE_256'] = "invalid_signature"
+
+    post webhooks_repo_updates_path, headers: invalid_headers, as: :json, params: payload
+    assert_response 500
+  end
```

Comment:
This should probably be 403 not a 500. Reserve 500s for actual unexpected errors. That allows us to then capture those via Bugsnag and fix them all.